File: src/Init/Data/Int/DivMod/Basic.js
export function lean_int_emod(a, b) {
  a = BigInt(a);
  b = BigInt(b);
  if (b === 0n) return a;
  let r = a % b;
  if (r < 0n) {
    r += (b < 0n ? -b : b);
  }
  return r;
}

export function lean_int_ediv(a, b) {
  a = BigInt(a);
  b = BigInt(b);
  if (b === 0n) return 0n;
  return (a - lean_int_emod(a, b)) / b;
}

export function lean_int_div_exact(a, b) {
  a = BigInt(a);
  b = BigInt(b);
  if (b === 0n) return false;
  return a % b === 0n;
}

export function lean_int_div(a, b) {
  a = BigInt(a);
  b = BigInt(b);
  if (b === 0n) return 0n;
  return a / b;
}

export function lean_int_mod(a, b) {
  a = BigInt(a);
  b = BigInt(b);
  if (b === 0n) return a;
  return a % b;
}

File: src/rust/gen_init_ffi/src/ffi/Init/Data/Int/DivMod/Basic.rs
use leanh_l1::datatypes::{LeanObject, LeanScalarArray, LeanStringObject};
// Generated stub file for Lean FFI imports
// Source: src/Init/Data/Int/DivMod/Basic.lean

pub unsafe fn lean_int_ediv(a: *mut LeanObject, b: *mut LeanObject) -> *mut LeanObject {
    unsafe { lean_nat_div(a, b) }
}

pub unsafe fn lean_int_emod(a: *mut LeanObject, b: *mut LeanObject) -> *mut LeanObject {
    unsafe { lean_nat_mod(a, b) }
}

pub fn lean_int_div_exact(_: *mut LeanObject, _: *mut LeanObject) -> *mut LeanObject {
    todo!("Stub for lean_int_div_exact");
}

pub fn lean_int_div(_: *mut LeanObject, _: *mut LeanObject) -> *mut LeanObject {
    todo!("Stub for lean_int_div");
}

pub fn lean_int_mod(_: *mut LeanObject, _: *mut LeanObject) -> *mut LeanObject {
    todo!("Stub for lean_int_mod");
}
